### ISSUE TITLE
nodejs: drop back to LTS branch

### ISSUE THREAD

**Issue by MingcongBai** [AOSC-Dev/aosc-os-abbs#575]
As it currently stands, the use of newest Node.js has rendered the following packages impossible to build successfully/bug-free from source:

- extra-editors/atom
- extra-editors/vscode

... And many more are expected to be added to the tree in the forseeable future. With Node.js 8.* series not in sight (probably not until the latter half of the year), to keep the maintenance going - for AMD64 (which is covered, generally, with binary packages), and for all of our ports supported by Node.js - it is most important that we drop back to the LTS branches for both Node.js (`extra-js/nodejs`) and NPM (included within the package `nodejs`) to their LTS counterparts, as described here:

- (Node.js) https://github.com/nodejs/LTS#lts-schedule
- (NPM) https://github.com/npm/npm/wiki/LTS

This will take effect with this cycle of updates... And shall I kindly ask that @LER0ever to make attempts to recover the packages listed above to source build? ... Once the drop is done.

**Comment by MingcongBai** [comment]
A vote is currently underway (at our Telegram group) over the question of whether to drop `extra-web/n1` for their recent changes that required users to pay for their syncing service.

**Comment by MingcongBai** [comment]
A vote is currently underway (at our Telegram group) over the question of whether to drop `extra-web/n1` for their recent changes that required users to pay for their syncing service.

**Comment by MingcongBai** [comment]
The poll is closed. `extra-web/n1` will be dropped. Updating original issue text.

**Comment by LionNatsu** [comment]
I think it will be better if you post them here for tracking that the (main) reason why these packages fail on building.

**Comment by MingcongBai** [comment]
> I think it will be better if you post them here for tracking that the (main) reason why these packages fail on building.

The reason is neither Atom nor VSCode has the intend to support the cutting edge Node.js versions... We were able to maintain some degree of upstream report until Node.js started their version jumps, over the course of last year, we started seeing these two packages...

- Started to come with issues not found with official binaries (VSCode could not install extensions or packages).

Then...

- Straight up failing to build, the build system and the packages installed (and/or bundled) with the applications themselves.

Another reason here, as mentioned above, is the expectation that there would be more and more Node (Electron) based applications, many of which will run into the same sort of issue: ARM used to get Atom and VSCode updates before all the bin-packing started - and will later spread to PPC64 and MIPS64el if their Node.js support finishes - the gap between package support will get bigger overtime if we do not at least conform with the upstream. We simply do not have the capacity to contact the upstream in such a large quantity, especially when some distributions could simply use official binaries.

**Comment by MingcongBai** [comment]
Not the best outcome I know. But we might have to deal with this now, unfortunately.

**Comment by MingcongBai** [comment]
(Maybe we needed a "dehancement" tag)

**Comment by LionNatsu** [comment]
As a 'drop back to LTS' issue, this should be closed. We are now is still on 6.10.2, however it's 6.10.3 now.

BTW, are there any changes about the situation from 7.5 to 7.10.0 (three months)?

**Comment by MingcongBai** [comment]
> BTW, are there any changes about the situation from 7.5 to 7.10.0 (three months)?

Please open another issue for this question, thanks...

Closing this one.
